File: leetcode/9. Palindrome Number/one.go
func isPalindrome(x int) bool {
	if x < 0 {
		 return false
	 }
	 copied := x
	 reversed := 0
	 
	 for  x >  0 {
		 reversed = reversed * 10 + x % 10
		 x /= 10
	 }
	 
	 
	 return copied == reversed
 }

File: leetcode/9. Palindrome Number/one.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
      
      if x < 0 {
        return false
      }
      
      let mut copied = x;
      let mut reversed:i32 = 0;
      while copied > 0 {
        reversed = reversed * 10 + copied % 10;
        copied /= 10;
      }
      reversed == x
        
    }
}